validator/src/commands/run/args/pub_sub_config.rs

```diff
@@ -72,21 +120,17 @@ pub(crate) fn args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
             .validator(is_parsable::<usize>)
             .default_value(DEFAULT_RPC_PUBSUB_WORKER_THREADS)
             .help("PubSub worker threads"),
-        Arg::with_name("rpc_pubsub_notification_threads")
-            .long("rpc-pubsub-notification-threads")
-            .requires("full_rpc_api")
-            .takes_value(true)
-            .value_name("NUM_THREADS")
-            .validator(is_parsable::<usize>)
-            .default_value_if(
-                "full_rpc_api",
-                None,
-                &DEFAULT_RPC_PUBSUB_NUM_NOTIFICATION_THREADS,
-            )
-            .help(
-                "The maximum number of threads that RPC PubSub will use for generating \
-                 notifications. 0 will disable RPC PubSub notifications",
-            ),
+        if TEST {
+            rpc_pubsub_notification_threads.default_value(&DEFAULT_TEST_RPC_PUBSUB_WORKER_THREADS)
+        } else {
+            rpc_pubsub_notification_threads
```

Comment:
nit: Can we do this `if-else` outside of the `vec![]`; just a bit of preference to avoid deeper nesting